mempool/mempool_test.go

```diff
@@ -135,18 +135,21 @@ func TestRestoreMempool(t *testing.T) {
 	require.Equal(t, 0, pool.Len())
 
 	// push multiple transactions to empty mempool (1,2,3)
+	transactions := make([]mempool.BroadcastedTransaction, 3)
 	for i := uint64(1); i < 4; i++ {
 		senderAddress := new(felt.Felt).SetUint64(i)
 		chain.EXPECT().HeadState().Return(state, func() error { return nil }, nil)
 		state.EXPECT().ContractNonce(senderAddress).Return(new(felt.Felt).SetUint64(0), nil)
-		require.NoError(t, pool.Push(&mempool.BroadcastedTransaction{
+		transaction := mempool.BroadcastedTransaction{
```

Comment:
Nitpick: Could this one be called `tx`? The difference of just an `s` between two vars with such a long name makes it a bit hard to differentiate